internal/cmd/base/delete.go

```diff
@@ -49,29 +51,50 @@ func (dc *DeleteCmd) CobraCommand(s state.State) *cobra.Command {
 	return cmd
 }
 
-// Run executes a describe command.
+// Run executes a delete command.
 func (dc *DeleteCmd) Run(s state.State, cmd *cobra.Command, args []string) error {
-	var cmdErr error
 
-	for _, idOrName := range args {
-		resource, _, err := dc.Fetch(s, cmd, idOrName)
-		if err != nil {
-			cmdErr = errors.Join(cmdErr, err)
-			continue
-		}
+	wg := sync.WaitGroup{}
+	wg.Add(len(args))
+	actions, errs :=
+		make([]*hcloud.Action, len(args)),
+		make([]error, len(args))
 
-		// resource is an interface that always has a type, so the interface is never nil
-		// (i.e. == nil) is always false.
-		if reflect.ValueOf(resource).IsNil() {
-			cmdErr = errors.Join(cmdErr, fmt.Errorf("%s not found: %s", dc.ResourceNameSingular, idOrName))
-			continue
-		}
+	for i, idOrName := range args {
+		i, idOrName := i, idOrName
+		go func() {
+			defer wg.Done()
```

Comment:
Is this complexity really needed? Most of the waiting time happens while waiting for the actions anyway?